From ef8778bf1d2bf3a59ae4455e9c9fb0ab28eeba63 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 2 Mar 2017 15:51:39 +0800
Subject: [PATCH 1741/5242] MLK-14314-3 pxp-v3: add 'need_yuv_swap' field

commit  669ee4cd7646f6fede42835abb394058cf3ba843 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add 'need_yuv_swap' field to 'pxp_proc_data' structure to
record the yuv formats which needs byte swap.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 28ce43b27faad915e93f47b438d23f4ebfe020b5)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/pxp/pxp_dma_v3.c |    7 +++----
 include/uapi/linux/pxp_dma.h |    3 +++
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/dma/pxp/pxp_dma_v3.c b/drivers/dma/pxp/pxp_dma_v3.c
index 32ca050..5096665 100644
--- a/drivers/dma/pxp/pxp_dma_v3.c
+++ b/drivers/dma/pxp/pxp_dma_v3.c
@@ -425,7 +425,6 @@ static void pxp_set_ctrl(struct pxps *pxp)
 	struct pxp_proc_data *proc_data = &pxp_conf->proc_data;
 	u32 ctrl;
 	u32 fmt_ctrl;
-	int need_swap = 0;   /* to support YUYV and YVYU formats */
 
 	/* Configure S0 input format */
 	switch (pxp_conf->s0_param.pixel_fmt) {
@@ -461,14 +460,14 @@ static void pxp_set_ctrl(struct pxps *pxp)
 		break;
 	case PXP_PIX_FMT_YUYV:
 		fmt_ctrl = BV_PXP_PS_CTRL_FORMAT__UYVY1P422;
-		need_swap = 1;
+		proc_data->need_yuv_swap = true;
 		break;
 	case PXP_PIX_FMT_VYUY:
 		fmt_ctrl = BV_PXP_PS_CTRL_FORMAT__VYUY1P422;
 		break;
 	case PXP_PIX_FMT_YVYU:
 		fmt_ctrl = BV_PXP_PS_CTRL_FORMAT__VYUY1P422;
-		need_swap = 1;
+		proc_data->need_yuv_swap = true;
 		break;
 	case PXP_PIX_FMT_NV12:
 		fmt_ctrl = BV_PXP_PS_CTRL_FORMAT__YUV2P420;
@@ -487,7 +486,7 @@ static void pxp_set_ctrl(struct pxps *pxp)
 	}
 
 	ctrl = BF_PXP_PS_CTRL_FORMAT(fmt_ctrl) |
-		(need_swap ? BM_PXP_PS_CTRL_WB_SWAP : 0);
+		(proc_data->need_yuv_swap ? BM_PXP_PS_CTRL_WB_SWAP : 0);
 	__raw_writel(ctrl, pxp->base + HW_PXP_PS_CTRL_SET);
 
 	/* Configure output format based on out_channel format */
diff --git a/include/uapi/linux/pxp_dma.h b/include/uapi/linux/pxp_dma.h
index 308bdd2..2cfa8e4 100644
--- a/include/uapi/linux/pxp_dma.h
+++ b/include/uapi/linux/pxp_dma.h
@@ -205,6 +205,9 @@ struct pxp_proc_data {
 	int rot_pos;
 	int yuv;
 
+	/* to support YUYV and YVYU formats */
+	bool need_yuv_swap;
+
 	/* Source rectangle (srect) defines the sub-rectangle
 	 * within S0 to undergo processing.
 	 */
-- 
1.7.9.5

